src/llv8-inl.h

```diff
@@ -84,6 +83,21 @@ inline int64_t HeapObject::GetType(Error& err) {
 }
 
 
+inline bool HeapObject::ISJSErrorType(Error& err) {
+  int64_t type = GetType(err);
+  if (type == v8()->types()->kJSErrorType) return true;
+
+  // NOTE (mmarchini): We don't have a JSErroType constant on Node.js v6.x,
+  // thus we try to guess if the object is an Error object by checking if its
+  // name is Error. Should work most of the time.
+  if (!JSObject::IsObjectType(v8(), type)) return false;
+  JSObject obj(this);
+  if (obj.GetTypeName(err) != "Error" || err.Fail()) return false;
+
+  return true;
```

Comment:
I think this line and the `if` statement above could be combined into a single `return` statement.